### PR TITLE
Flitering

### PR DESCRIPTION
**Pull request by lepatrick714**
Review for completing the settings fragment

- [x] Update email 
- [x] Update password 
- [x] Update podcastLength

### CODE REVIEW

**Comment by ttony678** [comment]
This is a lot of code. Can we test this tomorrow with the server on?

**Review by ttony678 — CHANGES_REQUESTED** [review]
Everything looks good. One request that doesn't seem too hard would be to override back button pressed on SettingsFragment to take the user to HomeFragment instead of closing the app.

**Comment by lepatrick714** [comment]
@ttony678 Changes to the back button has been implemented.

**Review by ttony678 — APPROVED** [review]

File: app/src/main/java/appricottsoftware/clarity/fragments/SettingFragment.java [hunk 1]
```diff
@@ -24,6 +49,213 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
...
+                }
+                else {
+                    Toast.makeText(getContext(), "Invalid email", Toast.LENGTH_LONG).show();
+                    return false;
+                }
+            }
+        });
+    }
+
+    private boolean validatePassword(String password) {
```
Comment (ttony678) [inline app/src/main/java/appricottsoftware/clarity/fragments/SettingFragment.java:119]: If we care enough, we should mandate that users meet a criteria for safe password checking. I don't care for that to be done on t hisbranch, but it should be done eventually.

File: app/src/main/res/layout/fragment_setting.xml [hunk 2]
```diff
@@ -1,13 +1,151 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout
-    xmlns:android="http://schemas.android.com/apk/res/android"
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="@color/colorPrimaryDark">
+    xmlns:app="http://schemas.android.com/apk/res-auto">
```
Comment (ttony678) [inline app/src/main/res/layout/fragment_setting.xml:7]: Should we let users delete their account eventually? Future feature to consider. Overall I approve this pull request as is.

Comment (lepatrick714) [reply app/src/main/res/layout/fragment_setting.xml:7]: I'll work on this on a separate branch.

**Pull request merged**